packages/taco/src/conditions/base/rpc.ts

```diff
@@ -11,7 +11,9 @@ export const rpcConditionSchema = z.object({
   conditionType: z.literal(RpcConditionType).default(RpcConditionType),
   chain: createUnionSchema(SUPPORTED_CHAIN_IDS),
   method: z.enum(['eth_getBalance', 'balanceOf']),
-  parameters: z.array(EthAddressOrUserAddressSchema),
+  parameters: z.array(
+    z.union([EthAddressOrUserAddressSchema, z.string(), z.number()]),
```

Comment:
Why not just `EthAddressOrUserAddressSchema`? Why also number or string?

PS. love how that typing reads.

Reply:
The second parameter to `eth_getBalance` is `blockNumer`, which is a string (as per JSON RPC schema). I thought that accepting a number here would be more intuitive. Does that make sense? Or should we expose strict interfaces (lifted straight from JSON RPC)?

Reply:
Oh right I got you.

But then we ideally want something like:
```
parameters = (EthAddressOrUserAddressSchema) OR (EthAddressOrUserAddressSchema, Union[string, int]])
```

So something like the following then (my zod is bad)?
```
z.array(
     z.union([
           EthAddressOrUserAddressSchema,
           z.array(
                EthAddressOrUserAddressSchema,
                z.union([
                     z.string(),
                     z.number()]
               ])
          )
   ])
)
```

If the above is the simplest way (perhaps not), might be cleaner to define a custom schema type somewhere else, similar to how we did `EthAddressOrUserAddressSchema`